### PR TITLE
[PATCH v6] SCTP checksums support

### PR DESCRIPTION
**Pull request by lumag**
Add support for setting and validating SCTP checksums.

### CODE REVIEW

File: helper/include/odp/helper/chksum.h [hunk 1]
```diff
@@ -189,6 +189,48 @@ static inline int odph_udp_chksum_verify(odp_packet_t odp_pkt)
...
+ * handle arbitrary packet segmentation, including cases where the segments
+ * are not 2 byte aligned, nor have a length that is a multiple of 2.  This
+ * function also can handle jumbo frames (at least up to 10K).
+ *
+ * @param  odp_pkt     Calculate and compare the chksum for this SCTP pkt,
+ *                     which can be over IPv4 or IPv6.
+ * @return             Returns < 0 upon an error.  Returns 1 upon no error and
+ *                     the incoming chksum field is 0 (disabled), else returns 0
+ *                     if the incoming chksum field is correct, else returns 2
+ *                     when the chksum field is incorrect.
```
Comment (Bill-Fischofer-Linaro) [inline helper/include/odp/helper/chksum.h]: Better to document these as individual @retvals, consistent with rest of ODP APIs. That way they are split out by Doxygen.

File: helper/include/odp/helper/chksum.h [hunk 2]
```diff
@@ -189,6 +189,48 @@ static inline int odph_udp_chksum_verify(odp_packet_t odp_pkt)
...
+ * handle arbitrary packet segmentation, including cases where the segments
+ * are not 2 byte aligned, nor have a length that is a multiple of 2.  This
+ * function also can handle jumbo frames (at least up to 10K).
+ *
+ * This function will insert the calculated CRC32-c checksum into the proper
+ * location in the SCTP header.
+ *
+ * @param  odp_pkt     Calculate and insert chksum for this SCTP pkt, which can
+ *                     be over IPv4 or IPv6.
+ * @return             0 upon success and < 0 upon failure.
```
Comment (Bill-Fischofer-Linaro) [inline helper/include/odp/helper/chksum.h]: Individual @retvals please

File: test/validation/api/pktio/parser.c [hunk 3]
```diff
@@ -381,6 +381,24 @@ static void parser_test_vlan_qinq_ipv4_udp(void)
 	odp_packet_free(pkt);
 }
 
+static void parser_test_ipv4_sctp(void)
```
Comment (Bill-Fischofer-Linaro) [inline test/validation/api/pktio/parser.c]: What about a corresponding `parser_test_ipv6_sctp()` routine?

**Comment by lumag** [comment]
Added IPv6 SCTP test

**Comment by JannePeltonen** [comment]
Unfortunately I do not have time to actually review these patches but at a very quick look it looks like they are probably doing the right thing and fix the bug I reported.

**Comment by Bill-Fischofer-Linaro** [comment]
Thanks, Janne. I think this should be good to go then and if problems arise we can address them as bug reports if and when they do.

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**